OpenRA.Game/Graphics/Vertex.cs

```diff
@@ -15,24 +15,19 @@ namespace OpenRA.Graphics
 	[StructLayout(LayoutKind.Sequential)]
 	public struct Vertex
 	{
-		// TODO Workaround for unused field warnings in mono 2.10
-		#pragma warning disable 414
-		float x, y, z, u, v;
-		float p, c;
-		#pragma warning restore
+		public readonly float X, Y, Z, U, V, P, C;
 
 		public Vertex(float2 xy, float u, float v, float p, float c)
-		{
-			this.x = xy.X; this.y = xy.Y; this.z = 0;
-			this.u = u; this.v = v;
-			this.p = p; this.c = c;
-		}
+			: this(xy.X, xy.Y, 0, u, v, p, c) { }
```

Comment:
I considered removing these, but we then need to add a bunch of intermediate variables and manually extract the components out of the float2 / float[3].  That makes the callers even uglier than they already are.